### ISSUE TITLE
[BUG] Archi 4.7 Saving  - Connection Relationship Errors

### ISSUE THREAD

**Issue by reed155** [archimatetool/archi#609]
[archi4_7Error.docx](https://github.com/archimatetool/archi/files/4471327/archi4_7Error.docx)

### Version of Archi
4.7 Early Access

### Archi Plug-ins
Archi Excel Plugin
Archi Scripting
Lightbox

### Operating System
Windows 10

### Expected Behaviour
Should save without error as in 4.6

### Actual Behaviour
Does not Save, presents an error.

### Steps to Reproduce the Behaviour
1. Edit model
2. Attempt to Save
3.

Please contact me privately of the error log if needed.

**Comment by Phillipus** [comment]
This suggests that somehow the connection does not have the correct relationship. Does it open and save OK in Archi 4.6? Try deleting the connection and re-adding a new one.

**Comment by Phillipus** [comment]
Archi 4.7 has more error reporting for cases like this, so the bad connection may still be there in 4.6 but just not have an error message. The only way I can be sure is to have a copy of the model file.

**Comment by reed155** [comment]
It opens and saves fine in 4.6.  I may be able to share a copy of the file privately with some anonymization.  Please let me know how to share it with you directly.

**Comment by Phillipus** [comment]
Please email it to info  at  archimatetool.com

**Comment by jbsarrodie** [comment]
FWIW, one colleague very recently had the same issue. After some investigation, the issue was already there at the first time, but Archi 4.6 was not able to detect it. Archi 4.7 as able and raised the message. I had to manually fix the .archimate file.

@reed155 Do you use coArchi, the collaboration plugin, with this model ? If yes, did you also used some jArchi script to change the type of some objects ?

**Comment by Phillipus** [comment]
>...but Archi 4.6 was not able to detect it. Archi 4.7 as able and raised the message.

@jbsarrodie That's correct, because 4.7's `ModelChecker` now checks that a connection's source and target diagram objects match the connection's relationship's source and target elements. This will not raise an error in 4.6. The cause could lie elsewhere - in jArchi perhaps.

**Comment by jbsarrodie** [comment]
> This will not raise an error in 4.6. The cause could lie elsewhere - in jArchi perhaps.

I suspect a model corruption after a successful git merge:
- two users works on a model which contains (between others) a relationship R from A to B
- User A create a new view V using R, then publish
- User B change R so that it now goes from A to C, then publish
- Result is a model in which a view V exists with a connection from A to B, this connection is related to relationship R which is between A and C !

I once thought about this edge case but then forgot about it. I think we should make sure when reloading the model that all connections are in sync with underlying relationship, and if not simply update (if new source and/or target are still on the same view) or delete them.

**Comment by Phillipus** [comment]
@jbsarrodie We should track this case on the coArchi issue tracker. But @reed155 has not listed coArchi in his list of plug-ins, so maybe not the case here?

**Comment by reed155** [comment]
The model I derived my test model from leveraged the coArchi Plugin (Archi 4.6).   And the case that @jbsarrodie  mentioned seems like it very well could have happened.  We have many users that merge to the same model using CoArchi.

**Comment by reed155** [comment]
FWIW, I did not use jArchi to change object types on this model.
